thrifty-compiler/src/main/kotlin/com/microsoft/thrifty/compiler/ThriftyCompiler.kt

```diff
@@ -118,7 +120,14 @@ import kotlin.system.exitProcess
  * representation.  Rather than each enum member containing its value, a single large
  * function mapping enums to values will be generated.  This works around some JVM class-size
  * limitations in some extreme cases, such as an enum with thousands of members.  This should
- * be avoided unless you know you need it.  Implies `--lang=kotlin`.
+ * be avoided unless you know you need it.  Implies `--lang=kotlin` and mutually-exclusive with
+ * `--kt-huge-enums`.
+ *
+ * `--kt-huge-enums` is optional.  When specified, generated enums will use a different
+ * representation. Instead of `enum class`, a hierarchy of enum-like classes will be generated
+ * that is syntactically equivalent to access from Kotlin  This should
+ * be avoided unless you know you need it.  Implies `--lang=kotlin` and mutually-exclusive with
+ * `--kt-big-enums`.
```

Comment:
It might be useful to say what the integer value controls.

Super minor: the word wrapping here looks a little off.